### PR TITLE
[JS->TS] Migrate regression_tests.js

### PR DESCRIPTION
**Pull request by bh2smith**
In partial fulfillment of #330. Note that one test has been moved into a more appropriate place and one of the lint ignores was acting strange on my side so I moved them to be more directly in line with there intended purpose.

**Test Plan:** Unit tests and linter.

### CODE REVIEW

File: test/utilities.ts [hunk 1]
```diff
@@ -53,9 +53,10 @@ export async function closeAuction(
 }
 
 export async function sendTxAndGetReturnValue<T>(
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   method: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
     sendTransaction: (...args: any[]) => Promise<string>;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (nlordell) [inline test/utilities.ts]: nit: This shouldn't be needed anymore if you merge with master:
https://github.com/gnosis/dex-contracts/blob/master/test/utilities.ts#L55-L67

**Review by nlordell — APPROVED** [review]
Awesome! Just one minor nit with the disabled lints - they should already be fixed in master.

File: test/utilities.ts (hunk 1 above)
Comment (bh2smith) [reply test/utilities.ts]: Awesome thanks. Was that recent? I just branch off master this morning (or so I thought)

Comment (nlordell) [reply test/utilities.ts]: It was from this morning.

**Pull request merged**